email_handler.py

```diff
@@ -465,6 +468,23 @@ def handle_reply(envelope, smtp: SMTP, msg: Message, rcpt_to: str) -> (bool, str
         handle_bounce(contact, alias, msg, user, mailbox_email)
         return False, "550 SL E6"
 
+    mailb: Mailbox = Mailbox.get_by(email=mailbox_email)
+    if ENFORCE_SPF_AVAILABLE and mailb.force_spf:
+        if msg['X-SimpleLogin-Client-IP']:
+            r = spf.check2(i=msg['X-SimpleLogin-Client-IP'], s=envelope.mail_from.lower(), h=None)
```

Comment:
The library `spf` seems to be created by Scott Kitterman, the same guy behind `policyd-spf` so it should be quite battle tested :).